packages/nuxt/src/app/composables/preview.ts

```diff
@@ -0,0 +1,65 @@
+import { reactive, readonly, toRef, toRefs } from 'vue'
+
+import { refreshNuxtData, useRoute, useRouter } from '#app'
+
+export interface PreviewOptions {
+  controls: boolean
+  tokenQueryName: string
+}
+
+interface PreviewState {
+  enabled: boolean
+  token: string | null
+}
+
+const previewQueryName = 'preview'
+const previewDefaultOptions: PreviewOptions = {
+  controls: false,
+  tokenQueryName: 'token'
+}
+
+const tokensMap = new Map<string, string | null>()
+
+let isPreviewEnabled: boolean
+let addedAfterNavigationCallback = false
+
+export function usePreviewMode (options?: PreviewOptions) {
+  options = { ...previewDefaultOptions, ...(options || {}) }
+
+  const preview = reactive<PreviewState>({
+    enabled: false,
+    token: null
+  })
```

Comment:
Is it worth using `useState` or otherwise making this shared between invocations of `usePreviewMode` in different components?

Reply:
I think `isPreviewEnabled` on `L23` should handle cross components invocations?, but `useState` would be beneficial for future server integrations.

Will soon refactor it 👍